notebooks/Emukit-task-spread-of-disease-with-seir-model.ipynb

```diff
@@ -0,0 +1,908 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Set up the gillespie model\n",
+    "seir_model = seir.SEIR(population_size, alpha_init, beta)\n",
+    "seir_gil = seir.SEIRGillespie(seir_model)\n",
+    "\n",
+    "# Prior over alpha. This prior will be integrated against in Task 1 and 2\n",
+    "alpha_prior = seir.GammaPrior(5., 1., 5.)\n",
+    "\n",
+    "# computes expected time occurance and height of the infection peak for a given alpha. \n",
+    "meanmax = seir.MeanMaxInfectionGillespie(seir_gil, num_gil, t_end, alpha_prior)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Task 1: Height of peak of infection\n",
+    "\n",
+    "The height of the peak is the expected maximal number of simultaneously infected individuals \n",
+    "\n",
+    "$$\n",
+    "\\mathbb{E}_a\\left[\\max_t N_I(t, \\alpha)\\right] =\\int_{\\Omega}\\max_t N_I(t, \\alpha)p(\\alpha)\\mathrm{d}\\alpha.\n",
+    "$$\n",
+    "\n",
+    "We use the Bayesian quadrature package of Emukit to emulate the integrand $\\max_tN_I(t, \\alpha)p(\\alpha)$. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Define user function $\\max_t N_I(t, \\alpha)p(\\alpha)$ and integral bounds \n",
+    "\n",
+    "We call the integrand $\\max_t N_I(t, \\alpha)p(\\alpha)$ `f_height_of_peak_weighted`  since it computes the height of the infection peak weighted with the prior on $\\alpha$. One call of `f_height_of_peak_weighted` will run `num_gil` Gillespie runs. The integral bounds have units $\\alpha$, i.e., units of infection time divided by recovery time; they define the domain $\\Omega = [lower\\_bounds, upper\\_bound]$. We choose `80` to be the upper bound since we assume that the infection rate will not be larger than 80 times the recovery rate.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# define the integrand \n",
+    "f_height_of_peak_weighted = seir.height_of_peak_weighted(meanmax)\n",
+    "\n",
+    "# define integral bounds\n",
+    "lb = 1. # lower integral bound\n",
+    "ub = 80. # upper integral bound"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Get initial observations\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The quadrature model needs a few random staring locations to get an initial calibration of its hyperparameters."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# number of initial points\n",
+    "N_init_task1 = 3\n",
+    "\n",
+    "# inital traning points: locations X_init sampled uniformly from [lower_bound, upper_bound]. \n",
+    "X_init_task1 = np.random.rand(N_init_task1, 1)*(ub-lb) + lb \n",
```

Comment:
Or you could user mode_free.RandomDesign

Reply:
Fair point, but I think I'll leave it as it is. Actually I should sample from the alpha prior, but since I do not integrate against it but rather also emulate it and integrate agains a uniform measure this seemed sensible. It might be worth thinking about random initial designs once we can integrate against non-trivial measures (see issue #130 ). The BQ method can then use the random design that corresponds to the measure we integrate against. 

Having said that, RandomDesign is not very a descriptive name. I assume it samples uniformly?, we might want to put that in the name.

Reply:
Fair enough, let's leave it. Random does sample uniformly.